grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -12,6 +12,30 @@ class ProductService {
 	ApiService apiService
 	PermissionService permissionService
 	SubscriptionService subscriptionService
+	final Random random = new Random()
+
+	List<Product> relatedProducts(String id) {
+		Product product = Product.get(id)
+		if (product == null) {
+			return new ArrayList<Product>()
+		}
+		// find Product.owner's other products
+		Set<Product> all = new HashSet<Product>(Product.findAllByOwner(product.owner))
+		// find other products from the same category
+		Set<Product> cat = new HashSet<Product>(Product.findAllByCategory(product.category))
+		all.addAll(cat)
+
+		final int max = 3 // max number of related products
```

Comment:
Give as query param, with some default value (e.g. 3)? Should have an enforced maximum (e.g. 10)